framework/Models/HybridModels/HybridModel.py

```diff
@@ -170,23 +156,16 @@ def initialize(self,runInfo,inputs,initDict=None):
       @ In, initDict, dict, optional, dictionary of all objects available in the step is using this model
       @ Out, None
     """
-    if isinstance(self.modelInstance, Models.Model):
-      self.raiseAnError(IOError, "HybridModel has already been initialized, and it can not be initialized again!")
-    self.modelInstance = self.retrieveObjectFromAssemblerDict('Model', self.modelInstance)
-    if self.modelInstance.type == 'Code':
-      codeInput = []
-      for elem in inputs:
-        if isinstance(elem, Files.File):
-          codeInput.append(elem)
-      self.modelInstance.initialize(runInfo, codeInput, initDict)
-
+    HybridModelBase.initialize(self,runInfo,inputs,initDict)
     self.targetEvaluationInstance = self.retrieveObjectFromAssemblerDict('TargetEvaluation', self.targetEvaluationInstance)
     if len(self.targetEvaluationInstance):
       self.raiseAWarning("The provided TargetEvaluation data object is not empty, the existing data will also be used to train the ROMs!")
       self.existTrainSize = len(self.targetEvaluationInstance)
     self.tempTargetEvaluation = copy.deepcopy(self.targetEvaluationInstance)
-    if self.modelInstance is None:
-      self.raiseAnError(IOError,'Model XML block needs to be inputted!')
+    if len(self.modelInstances) != 1:
+      self.raiseAnError(IOError, 'Required one "Model" XML subnode under node "HybridModel" can be accepted!',
+                        '"{}" "Model" subnodes are provided.'.format(len(self.modelInstances)))
+    self.modelInstance = list(self.modelInstances.values())[0]
```

Comment:
Does the input specs help this list(entry)[0] so you don't need to do this? If you've already assured it's a length-one list, can you avoid recasting it as a list?

Reply:
First, it seems to me we can not check the number of each nodes via input specs. Second, for this case, the Model node is added in the base class, only HybridModel requires single model.